test/spec_utils.rb

```diff
@@ -257,30 +257,38 @@ def initialize(*)
     Rack::Utils.build_nested_query('foo' => 'bar', 'baz' => {}).
       should.equal 'foo=bar'
 
-    # The ordering of the output query string is unpredictable with 1.8's
-    # unordered hash. Test that build_nested_query performs the inverse
-    # function of parse_nested_query.
-    [{"foo" => nil, "bar" => ""},
-      {"foo" => "bar", "baz" => ""},
-      {"foo" => ["1", "2"]},
-      {"foo" => "bar", "baz" => ["1", "2", "3"]},
-      {"foo" => ["bar"], "baz" => ["1", "2", "3"]},
-      {"foo" => ["1", "2"]},
-      {"foo" => "bar", "baz" => ["1", "2", "3"]},
-      {"x" => {"y" => {"z" => "1"}}},
-      {"x" => {"y" => {"z" => ["1"]}}},
-      {"x" => {"y" => {"z" => ["1", "2"]}}},
-      {"x" => {"y" => [{"z" => "1"}]}},
-      {"x" => {"y" => [{"z" => ["1"]}]}},
-      {"x" => {"y" => [{"z" => "1", "w" => "2"}]}},
-      {"x" => {"y" => [{"v" => {"w" => "1"}}]}},
-      {"x" => {"y" => [{"z" => "1", "v" => {"w" => "2"}}]}},
-      {"x" => {"y" => [{"z" => "1"}, {"z" => "2"}]}},
-      {"x" => {"y" => [{"z" => "1", "w" => "a"}, {"z" => "2", "w" => "3"}]}}
-    ].each { |params|
-      qs = Rack::Utils.build_nested_query(params)
-      Rack::Utils.parse_nested_query(qs).should.equal params
-    }
```

Comment:
> detect bugs that might be in parse_nested_query-

You mean `build_nested_query`?

> Maybe we can add one test case that documents how build_nested_query and parse_nested_querywork together if we want to be explicit.

That sounds good to me.  We have some issue where `build` and `parse` don't play well together (#768), so having explicit "round tripping" tests would be good.

> Anyway, I can remove that commit from the PR if you think that is not worth it and just keep the other one :smiley:

TBH I'm mainly worried about being coupled to the internals.  Maybe someday we want to change the algorithm for the function and don't want to be coupled to hash ordering.  But maybe I shouldn't worry about that.  ;)

I think if we keep this, then just continue to add round tripping tests going forward, that would probably be best.